### DISCUSSION TITLE
How to represent grams per square metre per day

### DISCUSSION THREAD

**Discussion by edmondchuc** [qudt/qudt-public-repo#129] [Q&A] [answered]
I understand in the [Unit Vocabularies Submission Guide](https://github.com/qudt/qudt-public-repo/wiki/Unit-Vocabulary-Submission-Guidelines) under Rule 4 - Hyphens it states that 

> **c) There can be only 1 -PER- per qname**. 

I was wondering if the rule only applies to the local name or to the creation of the unit of measure as well?

We would like to create _grams per square metre per day_.

```
unit:GM-PER-M2-PER-DAY
    a qudt:Unit ;
    qudt:conversionMultiplier "1"^^xsd:double ;
    qudt:conversionOffset "0.0"^^xsd:double ;
    qudt:plainTextDescription "A metric unit of volume over time indicating the amount generated across one square meter over a day. Used to express productivity of an ecosystem." ;
    qudt:hasQuantityKind quantitykind:AreaPerTime, quantitykind:Mass ;
    qudt:symbol "g/m^2/day" ;
    rdfs:label "grams per square metre per day" ;
.
```

Can we still have something like the above if we change the local name to remove the double up of -PER-?

If the above is not valid, please clarify how we should proceed.

Thanks in advance.

**Comment by steveraysteveray** [comment, marked as answer]
You can indeed define the unit you want.
How it works is that everything after "-PER-" is in the denominator. Everything before "-PER-" is in the numerator. So the correct qname for your example is GM-PER-M2-DAY, which means:

```
GM
-------------
M**2  *  DAY
```

The natural language usage of these compound units is really somewhat ambiguous, since one could interpret "-PER-M2-PER-DAY" as either 1/(M2*DAY), or DAY/M2. Our naming rules remove that ambiguity.

**Comment by steveraysteveray** [comment]
I should add that the conversion multiplier in your example is incorrect. The multiplier should convert all the units to the SI equivalent, which in this case would be

```
kiloGM
-------------
M**2    *   SEC
```

So the conversion would be

```
1e-3
------------------    =  1.1574073e-8
1 *  24 * 3.6e3
```

**Comment by steveraysteveray** [comment]
Also, the dimensionality of your unit is:
```
Mass
--------------------
Length**2   *   Time
```
I did a quick check on what points to qkdv:A0E0L-2I0M1H0T-1D0 and found that we already have a quantitykind of quantitykind:MassPerAreaTime
That's the one you should be pointing to.

...which makes sense. In fact, the units that point to quantitykind:MassPerAreaTime include unit:KiloGM-PER-SEC-M2.
